src/nsrt_learning/side_predicate_learning.py

```diff
@@ -394,59 +400,92 @@ def _sideline(self) -> List[PartialNSRTAndDatastore]:
                 segment = seg_traj[t]
                 option = segment.get_option()
                 # Find the PNAD associated with this option.
-                pnad = param_opt_to_pnad[option.parent]
+                if param_opt_to_nec_pnad.get(option.parent) is None:
```

Comment:
can we initialize the `param_opt_to_nec_pnad` to be an empty list for each option? so we don't have to use this `.get()` stuff anywhere, which is hard for the reader to follow what's going on